File: 2920_melody/melody.cpp
#include<iostream>
#define debug
#define FILENAME "2920_melody/melody_in.txt"
#define FILE_CIN
using namespace std;
int main(){
#ifdef FILE_CIN
	freopen(FILENAME,"r",stdin);
	int tmp;
	cin>>tmp;
	for(int z=0;z<tmp;z++){
#endif
		int tt[8]={0,};
		for(int i=0;i<8;i++)
			cin>>tt[i];
		int i=0;
		string status="";//0 ascending, 1 descending, 2 mixed
		while(i<7){
			int t=tt[i];
			if(i==0){
				if(t-tt[i+1]==1)
					status="descending";
				else if(t-tt[i+1]==-1)
					status="ascending";
				else
					status="mixed";
			}
			else{
				if(status=="descending")
					if(t-tt[i+1]!=1){
						status="mixed";
						break;
					}
				if(status=="ascending")
					if(t-tt[i+1]!=-1){
						status="mixed";
						break;
					}
			}
			i++;
		}
		cout<<status<<endl;
#ifdef FILE_CIN
	}
#endif
	return 0;
}

File: b_2920_melody/melody.py
import os
import sys
file_cin=True
if file_cin:
    sys.stdin=open("2920_melody/melody_in.txt",'r')
tmp=1
if file_cin:
    tmp=input()
    tmp=int(tmp)
for i in range(0,tmp):
    li=list(map(int,input().split()))
    j=0
    answer=""
    while j<7:
        if j==0:
            if li[j]-li[j+1]==1:
                answer="descending"
            elif li[j]-li[j+1]==-1:
                answer="ascending"
            else:
                answer="mixed"
                break
        else:
            if answer=="descending":
                if li[j]-li[j+1]!=1:
                    answer="mixed"
                    break
            if answer=="ascending":
                if li[j]-li[j+1]!=-1:
                    answer="mixed"
                    break
        j+=1
    print(answer)